include/eve/module/core/regular/impl/convert.hpp

```diff
@@ -139,4 +126,179 @@ namespace eve::detail
       }
     }
   }
+
+  //================================================================================================
+  // wide<->wide convert
+
+  //================================================================================================
+  // wide<->wide default convert implementation
+  template<typename In, typename Out>
+  EVE_FORCEINLINE auto convert_impl(EVE_SUPPORTS(cpu_), In const &v0, as<Out> const &tgt) noexcept
+  {
+    using out_t = as_wide_t<Out, cardinal_t<In>>;
+
+    if constexpr(has_aggregated_abi_v<In>)
+    {
+      // If input is aggregated, we can slice and combine without lose of performance
+      return out_t{eve::convert(v0.slice(lower_),tgt),eve::convert(v0.slice(upper_),tgt)};
```

Comment:
I am not so sure about this. 
Consider converting from two registers of shorts to chars.
On x86 there are some two register shuffles that can do it, if I am not mistaken.

Reply:
The way the x86 impl is written, we can catch them over there and indeed, this is a case I don't handle yet.